### ISSUE TITLE
Generating 'iot-rest-api-server' fails with iotivity-node 1.2.0-1

### ISSUE THREAD

**Issue by gvancuts** [01org/iot-rest-api-server#49]
Generating the `iot-rest-api-server` fails for me since the upgrade requiring `iotivity-node` 1.2.0-1` or higher. THe error shown on the screen is the following:
```
gvancuts@gvancuts-nuc:~/iot-rest-api-server$ snapcraft 
Preparing to pull meta-iot-web 
Pulling meta-iot-web 
From https://github.com/01org/iot-rest-api-server
 * branch            master     -> FETCH_HEAD
Already up-to-date.
Downloading 'node-v4.4.4-linux-x64.tar.gz'[===========================================================================================================================================================] 100%
npm --cache-min=Infinity install
npm ERR! Linux 4.4.0-59-generic
npm ERR! argv "/home/gvancuts/iot-rest-api-server/parts/meta-iot-web/install/bin/node" "/home/gvancuts/iot-rest-api-server/parts/meta-iot-web/install/bin/npm" "--cache-min=Infinity" "install"
npm ERR! node v4.4.4
npm ERR! npm  v2.15.1
npm ERR! code ETARGET

npm ERR! notarget No compatible version found: iotivity-node@'>=1.2.0-1 <2.0.0'
npm ERR! notarget Valid install targets:
npm ERR! notarget ["0.9.2-0","0.9.2-1","1.0.0-0","1.0.0-1","1.0.0-2","1.0.0-3","1.0.0-4","1.0.1-0","1.0.1-1","1.1.0-1","1.1.0-2","1.1.0-3","1.1.0-4","1.1.0-5","1.1.1-0","1.1.1-1","1.1.1-2","1.1.1-3","1.2.0-0"]
npm ERR! notarget 
npm ERR! notarget This is most likely not a problem with npm itself.
npm ERR! notarget In most cases you or one of your dependencies are requesting
npm ERR! notarget a package version that doesn't exist.
npm ERR! notarget 
npm ERR! notarget It was specified as a dependency of 'iot-rest-api-server'
npm ERR! notarget 

npm ERR! Please include the following file with any support request:
npm ERR!     /home/gvancuts/iot-rest-api-server/parts/meta-iot-web/src/npm-debug.log
Command '['/bin/sh', '/tmp/tmpmchipfnc', 'npm', '--cache-min=Infinity', 'install']' returned non-zero exit status 1
```
My environment is Ubuntu 16.04. I have tried to do a `snapcraft clean` and also deleting some of snapcraft folders (`node_modules`, `stage` and `parts`) but I'm still seeing this problem. The full log is attached. It seems like `npm` uses some cache and does not refresh the list of `iotivity-node` packages available in the npm registry... but I don't know why.
[npm-debug.txt](https://github.com/01org/iot-rest-api-server/files/739249/npm-debug.txt)

**Comment by gvancuts** [comment]
I have finally solved this myself on my system. For the record, what solved it was to call `npm update` on the system. `npm outdated` showed what was causing the problem:
```
gvancuts@gvancuts-nuc:~/iot-rest-api-server$ npm outdated
Package         Current  Wanted    Latest    Location            
gulp           MISSING          3.9.1    3.9.1    iot-rest-api-server
gulp-nodemon   MISSING          2.2.1    2.2.1    iot-rest-api-server
iotivity-node  MISSING          1.2.0-1  1.2.0-1  iot-rest-api-server
undefined
```

I believe that the `nodejs` plugin in `snapcraft` could be improved in this regard as it calls `npm install` with the `--cache-min=Infinty` unconditionally, see [snapcraft nodejs plugin](https://github.com/snapcore/snapcraft/blob/master/snapcraft/plugins/nodejs.py#L133). I have raised that point on the snapcraft mailing list [here](https://lists.ubuntu.com/archives/snapcraft/2017-January/002821.html)
